### ISSUE TITLE
gef initialization when attaching to process

### ISSUE THREAD

**Issue by repnzscasb** [hugsy/gef#278]
Your issue will be closed unless you confirm the following:
  * [x] Did you use the latest version of GEF from `master` branch?
  * [x] Did you read the [documentation](https://gef.readthedocs.org/en/latest/) first?
  * [x] Did you check [closed issues](https://github.com/hugsy/gef/issues)?

### Step 1: Describe your environment

  * Operating System: linux
  * Architecture: probably any (tested on armv7 and x64)
  * GDB version (including the Python library version): gdb 7.11.1 + python 3.5 (on ubuntu x64), and gdb 8.1 + python 2.7

### Step 2: Describe your problem

When attaching to a running process without a .gdbinit file, gef seems to miss some initialization steps.

#### Steps to reproduce

  1. sleep 10000 &
  2. gdb -n -p $!
  3. source gef/gef.py
  4. context

#### Observed Results

Output of context:

```
gef➤  context
[ Legend: Modified register | Code | Heap | Stack | String ]
...
[!] Command 'context' failed to execute properly, reason: 'NoneType' object has no attribute 'all_registers'
```

Note that to fix this, it's enough to issue:

  1. python set_arch()
  2. context

#### Expected results

The usual "context" output.

#### Traces
```
Detailed stacktrace
↳ File "gef/gef.py", line 6436, in context_regs()
    →     regs = set([x.strip() for x in current_arch.all_registers])
↳ File "gef/gef.py", line 6399, in do_invoke()
    →     self.layout_mapping[section]()
↳ File "gef/gef.py", line 1978, in wrapper()
    →     return f(*args, **kwargs)
↳ File "gef/gef.py", line 3331, in invoke()
    →     self.do_invoke(argv)
```

**Comment by hugsy** [comment]
This is because when we attaching and then call `source gef.py`, the handlers (most notably [`new_objfile_handler`](https://github.com/hugsy/gef/blob/adb75fc4b54ca5000335d3a02c8aeb7035192206/gef.py#L2393)) is not executed.The easiest fix is to force a call to `new_objfile_handler` when `gef` is loaded. Since the other handlers are not required at that time, they won't be an issue.

**Comment by hugsy** [comment]
Fixed by 5825a1a

Thanks @repnzscasb !

**Comment by adarshkr532** [comment]
Hi, I am new to gdb.
I am using the below source code.
````
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>
#include <string.h>

int main(int argc, char **argv)
{
  char buffer[64];

  gets(buffer);
}
````
When the input overflows during debugging, I receive the error
`Command 'context' failed to execute properly, reason: 'NoneType' object has no attribute 'cast'`
I cannot figure out how to force a call to `new_objfile_handler` when `gef` is loaded.
Any help would be appreciated. Thanks
